cl/search/tests.py

```diff
@@ -2772,6 +2773,16 @@ def rebuild_index(self):
         # -f rebuilds index without prompt for confirmation
         call_command("search_index", "--rebuild", "-f")
 
+    @classmethod
+    def restart_rate_limit(self):
+        """Restart the rate limiter counter to avoid getting blocked in
+        frontend after  tests
+        """
+        r = make_redis_interface("CACHE")
+        keys = r.keys(":1:rl:*")
+        if keys:
+            r.delete(*keys)
```

Comment:
This feels like it should be a decorator or part of our default test class or something, no? It seems like the kind of nuisance that we'll need over and over again. Is there a nice way to make it happen by default in ALL tests, except where we are testing the rate limiters?

Reply:
Sure! I've added this method as a mixing `RestartRateLimitMixin` and applied it to all the CL test cases except for `SimpleTestCase` which is commonly used for tests that don't require requests.

This mixin will perform the `rate_limit` clean up on `tearDownClass` which will be applied automatically if the TestClass doesn't overrides the `tearDownClass` in these cases will be needed to call `super().tearDownClass()`.
I confirmed that classes that override `tearDownClass` are doing call the super() `tearDownClass`.

I looked for tests testing the rate limiters. I just found `TestRateLimiters` which only tests the parsing rates. Do we have any other tests that actually check the `rate_limit` on requests? In any case, I don't think this mixin affects them since is applied after all tests in the class are finished.